db/view/view_building_worker.cc

```diff
@@ -0,0 +1,516 @@
+        co_await std::move(state_observer);
+    }
+    co_await _state.clear_state();
+    co_await uninit_messaging_service();
+}
+
+future<> view_building_worker::stop() {
+    return drain();
+}
+
+future<> view_building_worker::run_view_building_state_observer() {
+    auto abort = _as.subscribe([this] () noexcept {
+        _vb_state_machine.event.broadcast();
+    });
+    try {
+        co_await _group0_client.wait_until_group0_upgraded(_as);
+    } catch (abort_requested_exception&) {
+        vbw_logger.debug("Got abort_requested_exception while waiting for group0 upgrade.");
+        co_return;
+    }
+
+    while (!_as.abort_requested()) {
+        bool sleep = false;
+        try {
+            vbw_logger.trace("view_building_state_observer() iteration");
+            auto read_apply_mutex_holder = co_await _group0_client.hold_read_apply_mutex(_as);
+
+            co_await update_building_state();
+            _as.check();
+
+            read_apply_mutex_holder.return_all();
+            co_await _vb_state_machine.event.wait();
+        } catch (abort_requested_exception&) {
+        } catch (broken_condition_variable&) {
+        } catch (...) {
+            vbw_logger.warn("view_building_state_observer failed with: {}", std::current_exception());
+            sleep = true;
+        }
+
+        if (sleep && !_as.abort_requested()) {
+            try {
+                vbw_logger.debug("Sleeping after exception.");
+                co_await seastar::sleep_abortable(std::chrono::seconds(1), _as);
+            } catch (...) {
+                vbw_logger.warn("view_building_state_observer sleep failed: {}", std::current_exception());
+            }
+        }
+    }
+}
+
+future<> view_building_worker::update_building_state() {
+    auto guard = co_await _group0_client.start_operation(_as, service::raft_timeout{});
+    co_await _state.update(*this);
+    co_await _state.finish_completed_tasks();
+}
+
+bool view_building_worker::is_shard_free(shard_id shard) {
+    return !std::ranges::any_of(_state.tasks_map, [&shard] (auto& task_entry) {
+        return task_entry.second->replica.shard == shard && task_entry.second->state == view_building_worker::batch_state::in_progress;
+    });
+}
+
+void view_building_worker::init_messaging_service() {
+    ser::view_rpc_verbs::register_work_on_view_building_tasks(&_messaging, [this] (std::vector<utils::UUID> ids) -> future<std::vector<service::view_building::view_task_result>> {
+        return container().invoke_on(0, [ids = std::move(ids)] (view_building_worker& vbw) -> future<std::vector<service::view_building::view_task_result>> {
+            return vbw.get_tasks_results(std::move(ids));
+        });
+    });
+}
+
+future<> view_building_worker::uninit_messaging_service() {
+    return ser::view_rpc_verbs::unregister(&_messaging);
+}
+
+future<std::vector<service::view_building::view_task_result>> view_building_worker::get_tasks_results(std::vector<utils::UUID> ids) {
```

Comment:
I would suggest changing the name of this method... Right now it is responsible for starting tasks as well, the `get_` prefix suggests that it's just a simple getter.

Can be changed in a follow-up.